bitnami/argo-cd/templates/repo-server/deployment.yaml

```diff
@@ -273,9 +309,82 @@ spec:
             - name: empty-dir
               mountPath: /tmp
               subPath: tmp-dir
+            {{- if .Values.repoServer.configManagementPlugins.enabled }}
+            # The sidecars register their sockets to this volume for the repo server to find them.
+            - name: cmp-server-plugins
+              mountPath: /home/argocd/cmp-server/plugins
+            {{- end }}
           {{- if .Values.repoServer.extraVolumeMounts }}
           {{- include "common.tplvalues.render" (dict "value" .Values.repoServer.extraVolumeMounts "context" $) | nindent 12 }}
           {{- end }}
+        {{- if .Values.repoServer.configManagementPlugins.enabled }}
+        {{- range $plugin := .Values.repoServer.configManagementPlugins.plugins }}
+        {{- $additionalBinariesDir := $plugin.additionalBinariesDir | default "/usr/local/bin" | trimSuffix "/" }}
+        - name: {{ $plugin.name }}
+          image: {{ include "common.images.image" (dict "imageRoot" $plugin.sidecar.image "global" $.Values.global) }}
```

Comment:
I think it would be better to simplify all this and let users decide what will be the sidecar preferred structure. It simplify our maintenance at the same time it allows total flexibility for users.

It would be the same approach that we follow in `.Values.repoServer.sidecars` for example.

Reply:
This block is kind of the whole point of this PR. There are some mounts that have to be mounted to specific directories, like the CMP binary, the `plugins` folder or the `plugin.yml`.

The user still has the option to completely set it by him or herself (through `.Values.repoServer.sidecars`). But removing this _opinionated_ block defeats the whole purpose of this PR.

Reply:
The thing is... This assumes a very specific object structure for each element in the `plugins` array. 
The main value the PR adds is automating the `volumeMounts` that are required for the sidecars to work. Could we sth like this instead?
```yaml
        {{- range $sidecar := .Values.repoServer.configManagementPlugins.sidecars }}
        {{- include "common.tplvalues.render" (dict "value" $sidecar "context" $) | nindent 8 }}
          volumeMounts:
            - name: cmp-{{ $sidecar.name }}
              mountPath: /home/argocd/cmp-server/config/plugin.yaml
              subPath: plugin.yaml
            - name: cmp-server-plugins
              mountPath: /home/argocd/cmp-server/plugins
            - name: cmp-additional-binaries
              mountPath: {{ $additionalBinariesDir }}
            - name: cmp-server-tmp-dir
              mountPath: /tmp
        {{- end }}
```
This way, we help users setting up mounted directories while we keep flexible enough for users to customize the sidecars.